src/store/DrawingCursorStore.js

```diff
@@ -0,0 +1,51 @@
+import { action, observable, when } from 'mobx';
+
+export default class DrawingCursorStore {
+    constructor(mainStore) {
+        this.mainStore = mainStore;
+        when(() => this.context, this.onContextReady);
+    }
+
+    get context() { return this.mainStore.chart.context; }
+    get stx() { return this.context.stx; }
+
+    @observable isDrawing;
+    @observable top;
+    @observable left;
+    @observable display;
+
+    onContextReady = () => {
+        this.stx.container.addEventListener('mousemove', this.onMouseMove);
+        this.stx.container.addEventListener('touchmove', this.onMouseMove);
+        this.stx.container.addEventListener('touchend', this.onTouchEnd);
+    };
+
+    @action.bound onMouseMove = (e) => {
+        e.preventDefault();
+        this.isDrawing = this.stx.currentVectorParameters.vectorType;
```

Comment:
as it explaind [here](http://documentation.chartiq.com/CIQ.Drawing.html#main) , Whenever CIQ.ChartEngine.vectorType==yourname, then your drawing tool will be the one that is enabled when the user begins a drawing. so i am using this property to know if we are in drawing mode or not.